Source/Shaders/BillboardCollectionFS.glsl

```diff
@@ -40,5 +68,26 @@ void main()
 #endif
 
     gl_FragColor = color;
+
     czm_writeLogDepth();
+
+#ifdef CLAMP_TO_GROUND
+    if (v_eyeDepth > -2000.0) {
+        vec2 adjustedST = v_textureCoordinates - v_textureOffset.xy;
+        adjustedST = adjustedST / vec2(v_textureOffset.z - v_textureOffset.x, v_textureOffset.w - v_textureOffset.y);
+
+        float globeDepth1 = getGlobeDepth(adjustedST, v_depthLookupTextureCoordinate1);
+        float globeDepth2 = getGlobeDepth(adjustedST, v_depthLookupTextureCoordinate2);
+        float globeDepth3 = getGlobeDepth(adjustedST, v_depthLookupTextureCoordinate3);
+
+        float epsilonEyeDepth = v_eyeDepth + czm_epsilon5;
+
+        // negative values go into the screen
+        if (globeDepth1 > epsilonEyeDepth && globeDepth2 > epsilonEyeDepth && globeDepth3 > epsilonEyeDepth)
```

Comment:
Make this a separate `if` statement after each call to `getGlobeDepth`.

On WebGL implementations with dynamic branches, it could exit early and avoid the memory bandwidth for the unnecessary texture reads.

Also, if not already done, order the calls to `getGlobeDepth` in the order of most likely to fail the depth test - I imagine origin first.

Reply:
I had it that way initially but @bagnell said it wouldn't work that way.  Is dynamic branches something that's only supported in certain browsers?

Reply:
On a lot of GPUs with WebGL 1 support, there is no dynamic branches, but @pjcozzi is right. Early exits for GPUs with dynamic branches would be faster.